lib/import.js

```diff
@@ -1,32 +1,30 @@
-var pump = require('pump')
+var miss = require('mississippi')
 var through = require('through2')
+var parseInputStream = require('parse-input-stream')
 
-module.exports = importStream
+module.exports = importPipeline
 
-function importStream (jawn, opts) {
+// Returns the parser at the beginning of the pipeline
+// When the pipeline finishes, it calls `callback(feedId, err)`.
+//   If there were no errors, err will be undefined.
+function importPipeline (jawn, opts, callback) {
   if (!opts) opts = {}
   var writeStream = jawn.core.createWriteStream(opts)
-  var stream = pump(parseInputStream(opts), writeStream, function done (err) {
-    stream.id = writeStream.id
-    console.log(err)
-  })
-  return stream
-}
+  var parser = parseInputStream(opts)
+  var transform = through.obj(stringifyData, end)
 
-// Transform input CSV, JSON, etc to json objects that will be written as blocks in our hypercore feed
-// TODO: Implement this! See https://github.com/CfABrigadePhiladelphia/jawn/issues/32
-// This is the functionality that parse-input-stream aims to support.
-// If you can get that module to work, feel free to use it!
-function parseInputStream (opts) {
-  var transformStream = through(write, end)
-  return transformStream
+  miss.pipe(parser, transform, writeStream, function done (err) {
+    callback(err, writeStream.id)
+  })
 
-  function write (buffer, encoding, next) {
-    // transform input CSV, JSON, etc. here ...
+  function stringifyData (data, enc, next) {
+    this.push(JSON.stringify(data))
     next()
   }
 
   function end (done) {
     done()
   }
+
+  return parser
```

Comment:
That's worth trying. I recommend merging this PR as-is so you can fork it
and submit a version that uses miss.pipeline.

On Saturday, March 19, 2016, benjaminettori <notifications@github.com>
wrote:

> In lib/import.js
> <https://github.com/CfABrigadePhiladelphia/jawn/pull/34#discussion_r56748558>
> :
>
> >      next()
> >    }
> >
> >    function end (done) {
> >      done()
> >    }
> > +
> > +  return parser
>
> If we use miss.pipeline() instead of miss.pipe(), could we not then return
> a stream from this method? If so, why would we prefer to use miss.pipe()?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/CfABrigadePhiladelphia/jawn/pull/34/files/30c0be0cc5ea5a3ef6512efd472aad1f66ac7b82#r56748558>
>